test/extensions/quic_listeners/quiche/quic_io_handle_wrapper_test.cc

```diff
@@ -56,18 +71,20 @@ TEST_F(QuicIoHandleWrapperTest, DelegateIoHandleCalls) {
   EXPECT_CALL(os_sys_calls_, sendmsg(fd, _, 0)).WillOnce(Return(Api::SysCallSizeResult{5u, 0}));
   wrapper_->sendmsg(&slice, 1, 0, /*self_ip=*/nullptr, *addr);
 
-  Network::IoHandle::RecvMsgOutput output(nullptr);
-  EXPECT_CALL(os_sys_calls_, recvmsg(fd, _, 0)).WillOnce(Invoke([](int, struct msghdr* msg, int) {
-    sockaddr_storage ss;
-    auto ipv6_addr = reinterpret_cast<sockaddr_in6*>(&ss);
-    memset(ipv6_addr, 0, sizeof(sockaddr_in6));
-    ipv6_addr->sin6_family = AF_INET6;
-    ipv6_addr->sin6_addr = in6addr_loopback;
-    ipv6_addr->sin6_port = htons(54321);
-    *reinterpret_cast<sockaddr_in6*>(msg->msg_name) = *ipv6_addr;
-    msg->msg_namelen = sizeof(sockaddr_in6);
-    return Api::SysCallSizeResult{5u, 0};
-  }));
+  Network::IoHandle::RecvMsgOutput output(1, nullptr);
+  EXPECT_CALL(os_sys_calls_, recvmsg(fd, _, messageTruncatedOption()))
```

Comment:
It's curious that you had to introduce messageTruncatedOption in the backport.  I wonder what's going on in the master branch.

Reply:
I was getting macos test failures and could not access log artifact. So, after scrutinizing changes introduced in this PR, I noticed that the test expected recvmsg with MSG_TRUNC, which on macos is not used. It fixed the macos test issue on this 1.13 branch. On master it clearly expects recvmsg to be called with MSG_TRUNC. But I am not sure if on master CI tests execute all unit tests for macos build or only integration ones.

Reply:
cc @ggreenway 

Yeah, I think that some tests being excluded from CI at HEAD is the best way to explain the difference in behavior in macos tests.

Reply:
Most tests aren't run in mac CI.

In fact, integration tests don't even build right now on mac.